src/neighbors.cc

```diff
@@ -1367,131 +1367,133 @@ neighborDumpPeers(StoreEntry * sentry)
 void
 dump_peer_options(StoreEntry * sentry, CachePeer * p)
 {
+    PackableStream os(*sentry);
+
     if (p->options.proxy_only)
-        storeAppendPrintf(sentry, " proxy-only");
+        os << " proxy-only";
 
     if (p->options.no_query)
-        storeAppendPrintf(sentry, " no-query");
+        os << " no-query";
 
     if (p->options.background_ping)
-        storeAppendPrintf(sentry, " background-ping");
+        os << " background-ping";
 
     if (p->options.no_digest)
-        storeAppendPrintf(sentry, " no-digest");
+        os << " no-digest";
 
     if (p->options.default_parent)
-        storeAppendPrintf(sentry, " default");
+        os << " default";
 
     if (p->options.roundrobin)
-        storeAppendPrintf(sentry, " round-robin");
+        os << " round-robin";
 
     if (p->options.carp)
-        storeAppendPrintf(sentry, " carp");
+        os << " carp";
 
 #if USE_AUTH
     if (p->options.userhash)
-        storeAppendPrintf(sentry, " userhash");
+        os << " userhash";
 #endif
 
     if (p->options.sourcehash)
-        storeAppendPrintf(sentry, " sourcehash");
+        os << " sourcehash";
 
     if (p->options.weighted_roundrobin)
-        storeAppendPrintf(sentry, " weighted-round-robin");
+        os << " weighted-round-robin";
 
     if (p->options.mcast_responder)
-        storeAppendPrintf(sentry, " multicast-responder");
+        os << " multicast-responder";
 
 #if PEER_MULTICAST_SIBLINGS
     if (p->options.mcast_siblings)
-        storeAppendPrintf(sentry, " multicast-siblings");
+        os << " multicast-siblings";
 #endif
 
     if (p->weight != 1)
-        storeAppendPrintf(sentry, " weight=%d", p->weight);
+        os << " weight=" << p->weight;
 
     if (p->options.closest_only)
-        storeAppendPrintf(sentry, " closest-only");
+        os << " closest-only";
 
 #if USE_HTCP
     if (p->options.htcp) {
-        storeAppendPrintf(sentry, " htcp");
+        os << " htcp";
         if (p->options.htcp_oldsquid || p->options.htcp_no_clr || p->options.htcp_no_purge_clr || p->options.htcp_only_clr) {
             bool doneopts = false;
             if (p->options.htcp_oldsquid) {
-                storeAppendPrintf(sentry, "oldsquid");
+                os << "oldsquid";
```

Comment:
This reporting is broken in the official code as well (missing `=`). I did not fix it, but I am not against adding this fix to this PR scope, especially if that fix is added in this form (that would improve code safety during refactoring and facilitate future code improvements discussed in another review comment):

```suggestion
                os << (doneopts ? ',' : '=') << "oldsquid";
```

Reply:
I ported like-for-like, but I agree with fixing this